### PR TITLE
Create historical versions of source statistics built in views

### PR DESCRIPTION
**Pull request by SangJunBak**
Due to an issue with latency and high CPU usage, we create a non-historical index + view on top of mz_source_statistics such that Console's source lists efficiently query on it. The latency and high CPU usage is related to us retaining history, causing the number of updates in the index to be large.

From 5 cases, there seems to be a stark improvement with these new indexes:

Screenshot 2024-05-16 at 2 16 11 PM

Screenshot 2024-05-16 at 2 19 18 PM

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/pgrepr-consts/src/oid.rs [hunk 1]
```diff
@@ -674,3 +674,5 @@ pub const INDEX_MZ_KAFKA_SOURCES_IND_OID: u32 = 16950;
 pub const INDEX_MZ_WEBHOOK_SOURCES_IND_OID: u32 = 16951;
 pub const TABLE_MZ_HISTORY_RETENTION_STRATEGIES_OID: u32 = 16952;
 pub const SOURCE_MZ_MATERIALIZED_VIEW_REFRESHES_OID: u32 = 16953;
+pub const VIEW_MZ_SOURCE_STATISTICS_OID: u32 = 16954;
```
Comment (maddyblue) [inline src/pgrepr-consts/src/oid.rs:677]: If this name is staying, it should keep its old oid. Give VIEW_MZ_SOURCE_STATISTICS_WITH_HISTORY_OID and the other new thing new oids instead.

**Review by RobinClowers — COMMENTED** [review]
I'm not sure how this works, but does renaming those indexes mean we will loose all the existing historical data?

**Review by maddyblue — COMMENTED** [review]
This will need signoff from the epd-mz-introspection-council due to a new index.

**Comment by SangJunBak** [comment]
>I'm not sure how this works, but does renaming those indexes mean we will loose all the existing historical data?

@RobinClowers My first guess would be no given the historical view derives from `MZ_SOURCE_STATISTICS_RAW` that has the existing historical data so this new index would hydrate with the same state

**Comment by chaas** [comment]
LGTM from mz_introspection council. The new index has only a very small number of records, so the memory impact should be minimal.

**Review by maddyblue — APPROVED** [review]
Good chance that the autogenerated/docs tests will fail due to the new things not being documented. `./ci/test/lint-docs-catalog.sh --rewrite` may help there, and probably need to add docs.

**Review by benesch — APPROVED** [review]

**Pull request merged**